filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -1526,30 +1524,19 @@ void VulkanDriver::readPixels(Handle<HwRenderTarget> src, uint32_t x, uint32_t y
     // Perform the blit.
 
     vkCmdCopyImage(cmdbuffer, srcTarget->getColor(mContext.currentSurface, 0).image,
-            VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL, stagingImage, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+            VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL, stagingImage, VK_IMAGE_LAYOUT_GENERAL,
```

Comment:
Personally, I would like a comment everywhere there is a hardcoded LAYOUT constant, explaining how/why that constant was chosen.

Reply:
I agree, added a comment here but I would like to do another pass in a future PR and make it even more clear.

Side note: another thing I don't like is that there is a lot of special case code to handle situations that involve the SwapChain (as opposed to a "normal" Texture), so I am considering adding a proxy VulkanTexture to VulkanSwapChain.  This might also allow us to consolidate all the layout logic into the VulkanTexture wrapper, rather than sprinkling it throughout the codebase.